packages/proto-signing/src/codec/cosmos/bank/v1beta1/tx.ts

```diff
@@ -0,0 +1,324 @@
+/* eslint-disable */
+import { Coin } from "../../../cosmos/base/v1beta1/coin";
+import { Input, Output } from "../../../cosmos/bank/v1beta1/bank";
+import { Reader, Writer } from "protobufjs/minimal";
+
+/**
+ *  MsgSend represents a message to send coins from one account to another.
+ */
+export interface MsgSend {
+  fromAddress: string;
+  toAddress: string;
+  amount: Coin[];
+}
+
+/**
+ *  MsgSendResponse defines the Msg/Send response type.
+ */
+export interface MsgSendResponse {}
+
+/**
+ *  MsgMultiSend represents an arbitrary multi-in, multi-out send message.
+ */
+export interface MsgMultiSend {
+  inputs: Input[];
+  outputs: Output[];
+}
+
+/**
+ *  MsgMultiSendResponse defines the Msg/MultiSend response type.
+ */
+export interface MsgMultiSendResponse {}
+
+const baseMsgSend: object = {
+  fromAddress: "",
+  toAddress: "",
+};
+
+const baseMsgSendResponse: object = {};
+
+const baseMsgMultiSend: object = {};
+
+const baseMsgMultiSendResponse: object = {};
+
+/**
+ *  Msg defines the bank Msg service.
+ */
+export interface Msg {
+  /**
+   *  Send defines a method for sending coins from one account to another account.
+   */
+  Send(request: MsgSend): Promise<MsgSendResponse>;
+
+  /**
+   *  MultiSend defines a method for sending coins from some accounts to other accounts.
+   */
+  MultiSend(request: MsgMultiSend): Promise<MsgMultiSendResponse>;
+}
+
+export class MsgClientImpl implements Msg {
+  private readonly rpc: Rpc;
+
+  constructor(rpc: Rpc) {
+    this.rpc = rpc;
+  }
+
+  Send(request: MsgSend): Promise<MsgSendResponse> {
+    const data = MsgSend.encode(request).finish();
+    const promise = this.rpc.request("cosmos.bank.v1beta1.Msg", "Send", data);
+    return promise.then((data) => MsgSendResponse.decode(new Reader(data)));
+  }
+
+  MultiSend(request: MsgMultiSend): Promise<MsgMultiSendResponse> {
+    const data = MsgMultiSend.encode(request).finish();
+    const promise = this.rpc.request("cosmos.bank.v1beta1.Msg", "MultiSend", data);
+    return promise.then((data) => MsgMultiSendResponse.decode(new Reader(data)));
+  }
+}
+
+interface Rpc {
+  request(service: string, method: string, data: Uint8Array): Promise<Uint8Array>;
+}
+
+export const protobufPackage = "cosmos.bank.v1beta1";
+
+export const MsgSend = {
+  encode(message: MsgSend, writer: Writer = Writer.create()): Writer {
+    writer.uint32(10).string(message.fromAddress);
+    writer.uint32(18).string(message.toAddress);
+    for (const v of message.amount) {
+      Coin.encode(v!, writer.uint32(26).fork()).ldelim();
+    }
+    return writer;
+  },
+  decode(input: Uint8Array | Reader, length?: number): MsgSend {
+    const reader = input instanceof Uint8Array ? new Reader(input) : input;
+    let end = length === undefined ? reader.len : reader.pos + length;
+    const message = { ...baseMsgSend } as MsgSend;
+    message.amount = [];
+    while (reader.pos < end) {
+      const tag = reader.uint32();
+      switch (tag >>> 3) {
+        case 1:
```

Comment:
Interesting, it checks the field number, but doesn't check if the type matches. (last 3 bytes)
So, if I encoded an int with field 1, this would not detect the mis-match.

(Or does this happen magically in `reader.string()`? I assume these calls consume bytes from the reader)